ui/src/app/pages/analysis/interactive-notebook.spec.tsx

```diff
@@ -0,0 +1,87 @@
+import { MemoryRouter, Route } from 'react-router-dom';
+
+import { AppStatus, WorkspaceAccessLevel } from 'generated/fetch';
+
+import { screen } from '@testing-library/dom';
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import * as swaggerClients from 'app/services/swagger-fetch-clients';
+import { currentWorkspaceStore } from 'app/utils/navigation';
+import { userAppsStore } from 'app/utils/stores';
+
+import { createListAppsRStudioResponse } from 'testing/stubs/apps-api-stub';
+import { workspaceStubs } from 'testing/stubs/workspaces';
+
+import { InteractiveNotebook } from './interactive-notebook';
+
+let mockNotebooksApi;
+
+beforeEach(async () => {
+  currentWorkspaceStore.next({
+    ...workspaceStubs[0],
+    accessLevel: WorkspaceAccessLevel.WRITER,
+  });
+
+  jest.mock('app/services/swagger-fetch-clients');
+  mockNotebooksApi = jest.spyOn(swaggerClients, 'notebooksApi');
+});
+
+const setup = (mockOverrides) => {
+  mockNotebooksApi.mockImplementation(() => ({
+    ...mockOverrides,
+  }));
+};
+
+const renderInteractiveNotebook = (pathParameters) =>
+  render(
+    <MemoryRouter
+      initialEntries={[
+        '/workspaces/sampleNameSpace/sampleWorkspace/notebooks/preview/example.Rmd',
+      ]}
+    >
+      <Route path='/workspaces/:ns/:wsid/notebooks/preview/:nbName'>
+        <InteractiveNotebook hideSpinner={() => {}} match={pathParameters} />
+      </Route>
+    </MemoryRouter>
+  );
+
+test('Edit Rmd file with running RStudio', async () => {
```

Comment:
React Testing Library does not seem to have functionality to check whether a new tab is opened. Checking whether a particular function is called (window.open) feels a bit like implementation details, however I see that it is also checking to see whether a certain behavior, opening a new window, has been observed. This seems like a bit of a gray area, so I welcome feedback.

Reply:
I had the same thoughts when testing the other open button, which has a very similar unit test. I think this behavior is best verified in an E2E test, which doesn't mock the browser. I think this is the best we can do at the unit test level, so I trust your judgement on whether this test is worthwhile or not.